src/platform/common/platform/fileSystem.node.ts

```diff
@@ -53,17 +53,7 @@ export class FileSystem extends FileSystemBase implements IFileSystemNode {
     }
 
     public async deleteLocalDirectory(dirname: string) {
-        const uri = vscode.Uri.file(dirname);
-        // The "recursive" option disallows directories, even if they
-        // are empty.  So we have to deal with this ourselves.
-        const files = await this.vscfs.readDirectory(uri);
-        if (files && files.length > 0) {
-            throw createDirNotEmptyError(dirname);
-        }
-        return this.vscfs.delete(uri, {
-            recursive: true,
-            useTrash: false
-        });
+        await new Promise((resolve) => fs.rm(dirname, { force: true, recursive: true }, resolve));
```

Comment:
Over engineered. We don't need to use VS Code API just to delete a folder.
We can just use native API for this.
VSC API is useful only when dealing with multiple file systems (such as remots or somethign that a user has contributed, i.e. when using Uris, even when we can easily check if uri.scheme=file and then revert to using node API which would be much faster)